src/test/java/com/cst438/JunitTestGradebook.java

```diff
@@ -0,0 +1,359 @@
+		course.setInstructor(TEST_INSTRUCTOR_EMAIL);
+		course.setEnrollments(new java.util.ArrayList<Enrollment>());
+		course.setAssignments(new java.util.ArrayList<Assignment>());
+
+		Enrollment enrollment = new Enrollment();
+		enrollment.setCourse(course);
+		course.getEnrollments().add(enrollment);
+		enrollment.setId(TEST_COURSE_ID);
+		enrollment.setStudentEmail(TEST_STUDENT_EMAIL);
+		enrollment.setStudentName(TEST_STUDENT_NAME);
+
+		Assignment assignment = new Assignment();
+		assignment.setCourse(course);
+		course.getAssignments().add(assignment);
+		// set dueDate to 1 week before now.
+		assignment.setDueDate(new java.sql.Date(System.currentTimeMillis() - 7 * 24 * 60 * 60 * 1000));
+		assignment.setId(1);
+		assignment.setName("Assignment 1");
+		assignment.setNeedsGrading(1);
+
+		AssignmentGrade ag = new AssignmentGrade();
+		ag.setAssignment(assignment);
+		ag.setId(1);
+		ag.setScore("80");
+		ag.setStudentEnrollment(enrollment);
+
+		// given -- stubs for database repositories that return test data
+		given(assignmentRepository.findById(1)).willReturn(Optional.of(assignment));
+		given(assignmentGradeRepository.findByAssignmentIdAndStudentEmail(1, TEST_STUDENT_EMAIL)).willReturn(ag);
+		given(assignmentGradeRepository.findById(1)).willReturn(Optional.of(ag));
+
+		// end of mock data
+
+		// then do an http get request for assignment 1
+		response = mvc.perform(MockMvcRequestBuilders.get("/gradebook/1").accept(MediaType.APPLICATION_JSON))
+				.andReturn().getResponse();
+
+		// verify return data with entry for one student without no score
+		assertEquals(200, response.getStatus());
+
+		// verify that a save was NOT called on repository because student already has a
+		// grade
+		verify(assignmentGradeRepository, times(0)).save(any());
+
+		// verify that returned data has non zero primary key
+		GradebookDTO result = fromJsonString(response.getContentAsString(), GradebookDTO.class);
+		// assignment id is 1
+		assertEquals(1, result.assignmentId);
+		// there is one student list
+		assertEquals(1, result.grades.size());
+		assertEquals(TEST_STUDENT_NAME, result.grades.get(0).name);
+		assertEquals("80", result.grades.get(0).grade);
+
+		// change grade to score = 88
+		result.grades.get(0).grade = "88";
+
+		// send updates to server
+		response = mvc
+				.perform(MockMvcRequestBuilders.put("/gradebook/1").accept(MediaType.APPLICATION_JSON)
+						.content(asJsonString(result)).contentType(MediaType.APPLICATION_JSON))
+				.andReturn().getResponse();
+
+		// verify that return status = OK (value 200)
+		assertEquals(200, response.getStatus());
+
+		// verify that repository save method was called
+		// AssignmentGrade must override equals method for this test for work !!!
+		AssignmentGrade updatedag = new AssignmentGrade();
+		updatedag.setId(1);
+		updatedag.setScore("88");
+		verify(assignmentGradeRepository, times(1)).save(updatedag);
+	}
+	
+	// testing PostMapping that creates new assignments
```

Comment:
Same as previous comment for GradeBookController.java, I would consider separating out the assignment tests into a JUnitTestAssignment.java file